spatial/vptree/vptree.go

```diff
@@ -0,0 +1,372 @@
+// Copyright ©2019 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package vptree
+
+import (
+	"container/heap"
+	"math"
+	"sort"
+
+	"golang.org/x/exp/rand"
+
+	"gonum.org/v1/gonum/stat"
+)
+
+// Comparable is the element interface for values stored in a vp tree.
+type Comparable interface {
+	// Distance returns the Euclidean distance between the receiver and
```

Comment:
If it's for general metric spaces, then the distance does not have to be Euclidean.

Reply:
Fixed